tests/utils/test_enumerablemap.py

```diff
@@ -0,0 +1,123 @@
+import pytest
+from starkware.starknet.testing.starknet import Starknet
+from starkware.starkware_utils.error_handling import StarkException
+from starkware.starknet.definitions.error_codes import StarknetErrorCode
+from utils import assert_revert, contract_path, TRUE, FALSE
+
+@pytest.fixture
+async def enumerablemap_mock():
+    starknet = await Starknet.empty()
+
+    enumerablemap = await starknet.deploy(
+        contract_path("tests/mocks/enumerablemap_mock.cairo")
+    )
+
+    return enumerablemap
+
+
+@pytest.mark.asyncio
+async def test_set(enumerablemap_mock):
+    enumerablemap = enumerablemap_mock
+
+    success = await enumerablemap.set(6, 8).invoke()
+    assert success.result.success == TRUE
+
+    executed_info = await enumerablemap.contains(6).call()
+    assert executed_info.result == (TRUE,)
+
+
+@pytest.mark.asyncio
+async def test_remove(enumerablemap_mock):
+    enumerablemap = enumerablemap_mock
+
+    success = await enumerablemap.set(6, 8).invoke()
+    assert success.result.success == TRUE
+
+    success = await enumerablemap.remove(6).invoke()
+    assert success.result.success == TRUE
+
+
+    executed_info = await enumerablemap.contains(6).call()
+    assert executed_info.result == (FALSE,)
+
+
+@pytest.mark.asyncio
+async def test_remove_fail(enumerablemap_mock):
+    enumerablemap = enumerablemap_mock
+
+    success = await enumerablemap.remove(6).invoke()
+    assert success.result.success == FALSE
+
+
+    executed_info = await enumerablemap.contains(6).call()
+    assert executed_info.result == (FALSE,)
```

Comment:
shouldn't this be called _before_ the `remove` to make sure it's not contained? it will always be presumably `FALSE` otherwise

Reply:
i think we should also test the length substraction